kornia/geometry/vector.py

```diff
@@ -0,0 +1,78 @@
+from typing import List
+
+from kornia.core import Module, Parameter, Tensor, normalize, rand
+from kornia.geometry.linalg import batched_dot_product
+from kornia.geometry.linalg import batched_squared_norm as _squared_norm
+from kornia.testing import KORNIA_CHECK_SHAPE
+
+__all__ = ["UnitVector"]
+
+
+# TODO: model _Scalar
+class _ScalarType:
+    pass
+
+
+# NOTE: not very clear if this is a module or custom Tensor
+# tensor subclass doesn't like to override methods
```

Comment:
Module, I guess

Reply:
We should explore this https://colab.research.google.com/drive/1WiuUjznGVOyDvn03bt8ZUbCtGS8lqOBN

@cjpurackal @YanivHollander

Reply:
/cc @stevenlovegrove for awareness— context is discussion in slack to model a BaseVectorType to mimic Eigen::Vector , Eigen::Matrix, etc to build on top of that things like Vec2, Vec3, RotationMatrix, HomographyMatrix, ProjectionMatrix, etc.

also one attempt is did some time ago to model a Image api, that maybe we could align with new Sophus Image or pangolin  : https://github.com/kornia/kornia/pull/1562

Only "drawback" is that will be pytorch only , while we have been recently exploring how to extend/adapt Kornia code to run given other backends like numpy/numba, triton, jax, etc